packages/eslint-plugin/tests/rules/no-unnecessary-type-parameters.test.ts

```diff
@@ -230,6 +211,10 @@ ruleTester.run('no-unnecessary-type-parameters', rule, {
     `,
     'declare function get(): void;',
     'declare function get<T>(param: T[]): T;',
+    'declare function factoryArray<T>(): Array<T>;',
```

Comment:
From https://github.com/typescript-eslint/typescript-eslint/issues/9529#issuecomment-2223224801:

> Treating array types in a parameter slot (not return type) as singular.

This is in a return type. Folks could reasonably want this like:

```ts
function factoryArray<T>(): Array<T> {
  console.log("Making an array.");
  return [];
}

const array = factoryArray<number>();
array.push(1);
array.push(2);
```

Reply:
Yep, that's why we allow it

Reply:
...huh, I don't remember why I wanted to comment here. Ignore me. 😄